utils/mwm.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+import argparse
+import logging
+import os
+import subprocess
+from string import Template
+from django.conf import settings
+
+logger = logging.getLogger(__name__)
+
+
+class OSMToMWM(object):
+    """
+    Thin wrapper around osmconvert to convert osm xml to mwm.
+    """
+
+    def __init__(self, osm=None, pbffile=None, mwmfile=None, debug=False):
+        self.osm = osm
+        if not os.path.exists(self.osm):
+            raise IOError('Cannot find raw OSM data for this task.')
+        self.pbffile = pbffile
+        self.mwmfile = mwmfile
+        if not self.pbffile:
+            # create pbf path from osm path.
+            root = self.osm.split('.')[0]
+            self.pbffile = root + '.pbf' 
+        if not self.mwmfile:
+            # create mwm path from osm path.
+            root = self.osm.split('.')[0]
+            self.mwmfile = root + '.mwm' 
+        self.debug = debug
+        mwm_root_path = settings.EXPORT_MWM_ROOT
+        self.cmd = Template('osmconvert $osm --out-pbf >$pbf')
+        self.cmd_mwm = Template( mwm_root_path + 'tools/unix/generate_mwm.sh $pbf')
+
+    def convert(self, ):
+        convert_cmd = self.cmd.safe_substitute({'osm': self.osm, 'pbf': self.pbffile})
+        if(self.debug):
+            print 'Running: %s' % convert_cmd
+        proc = subprocess.Popen(convert_cmd, shell=True, executable='/bin/bash', stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        (stdout, stderr) = proc.communicate()
+        returncode = proc.wait()
+        if (returncode != 0):
+            logger.error('%s', stderr)
+            raise Exception, "osmconvert failed with return code: {0}".format(returncode)
+        if(self.debug):
+            print 'Osmconvert returned: %s' % returncode
+        convert_cmd_mwm = self.cmd_mwm.safe_substitute({'pbf': self.pbffile})
+        if(self.debug):
+            print 'Running: %s' % convert_cmd_mwm
+        proc_mwm = subprocess.Popen(convert_cmd_mwm, shell=True, executable='/bin/bash', stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        (stdout, stderr) = proc_mwm.communicate()
+        returncode = proc_mwm.wait()
```

Comment:
`returncode = proc.returncode` or just use the latter in `if` later, since `wait()` call is indirectly called by `communicate()`.